src/main/java/MyArrayList.java

```diff
@@ -0,0 +1,335 @@
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Objects;
+import java.util.Spliterator;
+import java.util.function.UnaryOperator;
+
+public class MyArrayList<T> implements List<T> {
+
+  private T[] tempTable;
+  private int size;
+
+  public MyArrayList() {
+    tempTable = (T[]) new Object[0];
+    size = 0;
+  }
+
+  public MyArrayList(Collection<? extends T> c) {
+
+    tempTable = (T[]) c.toArray();
+    size = tempTable.length;
+  }
+
+  public int size() {
+    return size;
+  }
+
+  public boolean isEmpty() {
+    return size == 0;
+  }
+
+  public boolean contains(final Object o) {
+    for (T element : tempTable) {
+      if (element.equals(o)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  public Object[] toArray() {
+    return Arrays.copyOf(tempTable, size);
+  }
+
+  public <T1> T1[] toArray(final T1[] a) {
+    if (a.length < size) {
+      return (T1[]) Arrays.copyOf(tempTable, size, a.getClass());
+    }
+    System.arraycopy(tempTable, 0, a, 0, size);
+    if (a.length > size) {
+      a[size] = null;
+    }
+    return a;
+  }
+
+  public boolean add(final T t) {
+    add(size, t);
+    return true;
+  }
+
+  public void add(final int index, final T element) {
+    Object[] createTable = new Object[size + 1];
+    System.arraycopy(tempTable, 0, createTable, 0, index);
+    createTable[index] = element;
+    System.arraycopy(tempTable, index, createTable, index + 1, size - index);
+    size++;
+    tempTable = (T[]) createTable;
+  }
+
+  public boolean remove(final Object o) {
+    int index = indexOf(o);
+    if (!(index == -1)) {
+      remove(index);
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  public T remove(final int index) {
+    Object[] createTable = new Object[size - 1];
+    System.arraycopy(tempTable, 0, createTable, 0, index);
+    System.arraycopy(tempTable, index + 1, createTable, index, (size - 1) - index);
+    size--;
+    T toRemove = tempTable[index];
+    tempTable = (T[]) createTable;
+    return toRemove;
+  }
+
+  public boolean removeAll(final Collection<?> c) {
+    Objects.requireNonNull(c);
+    boolean result = false;
+    for (Object o : c) {
+      while (this.contains(o)) {
+        this.remove(o);
```

Comment:
this is inefficient - please note that you do. O(n^2) - it can. be done in O(n) - iterate once over array. and mark all elements to remove as null, then. consilidate array by removing nulls.